.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -531,29 +539,37 @@ private DateTimeResolutionResult ParseRelativeUnit(string text, DateObject refer
 
                 DateObject beginTime;
                 var endTime = beginTime = referenceTime;
-
+                var ptTimex = string.Empty;
                 if (Config.UnitMap.ContainsKey(srcUnit))
                 {
                     switch (unitStr)
                     {
+                        case "D":
+                            endTime = DateObject.MinValue.SafeCreateFromValue(beginTime.Year, beginTime.Month, beginTime.Day);
+                            endTime = endTime.AddDays(1).AddSeconds(-1);
+                            ptTimex = "PT" + (endTime.Hour - beginTime.Hour) + "H";
```

Comment:
the duration is calculated by hour? suppose the reference time is 18:30:30, what will the rest of the day be interpreted?